mindmeld/components/question_answerer.py

```diff
@@ -451,803 +528,2178 @@ def _generate_mapping_data(embedder_model, embedding_fields):
+            calling `update_resource()` in FieldResource.
+
+            Use this method only to obtain metadata.
+            """
+
+            cache_path = self._get_index_cache_path(index_name)
+
+            if index_name in self:
+                metadata_dump = self._create_metadata_for_dumping(index_name)
+
+            elif os.path.exists(cache_path):
+                opfile = open(cache_path, "rb")
+                metadata_dump = pickle.load(opfile)
+                metadata_dump.pop("__all_ids")
+                opfile.close()
+
+            else:
+                return {}
+
+            # create field resource instance for each of the metadata
+            index_resources = {}
+            for field, cache_object in metadata_dump.items():
+                field_resource = (
+                    NonElasticsearchQuestionAnswerer.FieldResource.from_metadata(cache_object)
                 )
+                index_resources.update({field: field_resource})
+            return index_resources
 
-                functions.append(
-                    {
-                        "filter": {
-                            "nested": {
-                                "path": self.syn_field,
-                                "query": {
-                                    "match": {
-                                        self.syn_field
-                                        + ".name.normalized_keyword": self.value
-                                    }
-                                },
-                            }
-                        },
-                        "weight": self.DEFAULT_EXACT_MATCH_BOOSTING_WEIGHT,
-                    }
+        def delete_index(self, index_name):
+
+            # clear field resources
+            if index_name in self:
+                del self._indices[index_name]  # free the pointer
+
+            # clear index dump cache path, if required
+            cache_path = self._get_index_cache_path(index_name)
+            if cache_path and os.path.exists(cache_path):
+                os.remove(cache_path)
+
+        def update_and_dump_index(self, index_name, index_resources, index_all_ids):
+
+            # update
+            self._indices.update({index_name: index_resources})
+            self._indices_all_ids.update({index_name: index_all_ids})
+
+            # create repo for dumping if required
+            cache_path = self._get_index_cache_path(index_name)
+            dir_name = os.path.dirname(cache_path)
+            if not os.path.exists(dir_name):
+                os.makedirs(dir_name)
+
+            # dump
+            metadata_dump = self._create_metadata_for_dumping(index_name)
+            metadata_dump.update({"__all_ids": index_all_ids})
+            opfile = open(cache_path, "wb")
+            pickle.dump(metadata_dump, opfile)
+            opfile.close()
+
+        def is_available(self, index_name):
+            if (
```

Comment:
maybe `return index_name in self._indices or os.path.exists(self._get_index_cache_path(index_name))`?